core/src/main/java/org/wildfly/plugin/core/DefaultContainerDescription.java

```diff
@@ -0,0 +1,139 @@
+                                        final String releaseVersion, final String launchType, final boolean isDomain) {
+        this.productName = productName;
+        this.productVersion = productVersion;
+        this.releaseVersion = releaseVersion;
+        this.launchType = launchType;
+        this.isDomain = isDomain;
+    }
+
+    @Override
+    public String getProductName() {
+        return productName;
+    }
+
+    @Override
+    public String getProductVersion() {
+        return productVersion;
+    }
+
+    @Override
+    public String getReleaseVersion() {
+        return releaseVersion;
+    }
+
+    @Override
+    public String getLaunchType() {
+        return launchType;
+    }
+
+    @Override
+    public boolean isDomain() {
+        return isDomain;
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder result = new StringBuilder(64);
+        result.append(productName);
+        if (productVersion != null) {
+            result.append(' ').append(productVersion);
+            if (releaseVersion != null) {
+                result.append(" (WildFly Core ").append(releaseVersion).append(')');
+            }
+        } else {
+            if (releaseVersion != null) {
+                result.append(' ').append(releaseVersion);
+            }
+        }
+        if (launchType != null) {
+            result.append(" - launch-type: ").append(launchType);
+        }
+        return result.toString();
+    }
+
+    /**
+     * Queries the running container and attempts to lookup the information from the running container.
+     *
+     * @param client the client used to execute the management operation
+     *
+     * @return the container description
+     *
+     * @throws IOException if an error occurs while executing the management operation
+     */
+    static DefaultContainerDescription lookup(final ModelControllerClient client) throws IOException {
+        final ModelNode op = Operations.createReadResourceOperation(new ModelNode().setEmptyList());
+        op.get(ClientConstants.INCLUDE_RUNTIME).set(true);
+        final ModelNode result = client.execute(op);
+        if (Operations.isSuccessfulOutcome(result)) {
+            final ModelNode model = Operations.readResult(result);
+            final String productName = getValue(model, "product-name", "WildFly");
+            final String productVersion = getValue(model, "product-version");
+            final String releaseVersion = getValue(model, "release-version");
+            final String launchType = getValue(model, "launch-type");
+            return new DefaultContainerDescription(productName, productVersion, releaseVersion, launchType, "DOMAIN".equalsIgnoreCase(launchType));
+        } else {
+            Logger.getLogger(DefaultContainerDescription.class).errorf("Failed to read the root resource: ", Operations.getFailureDescription(result));
```

Comment:
I guess that `UNKNOWN` will be useless in most situations. Why not throwing an exception instead of returning `UNKNOWN`?